### ISSUE TITLE
Should the helperText have a value by default?

### ISSUE THREAD

**Issue by oliviertassinari** [mui-org/material-ui-pickers#1749]
## Environment

| Tech                 | Version |
| -------------------- | ------- |
| @material-ui/pickers | v4.0.0-alpha.6 |

## Steps to reproduce

```jsx
function App() {
  const [value, setValue] = React.useState(null);

  return (
    <LocalizationProvider dateAdapter={DateFnsUtils}>
      <DatePicker
        value={value}
        onChange={newValue => {
          setValue(newValue);
        }}
        renderInput={props => <TextField {...props} />}
      />
    </LocalizationProvider>
  );
}
```

https://codesandbox.io/s/condescending-hoover-m5jk7?file=/index.js:234-580

## Expected behavior

Should the helperText have a value by default? It feels strange. I think that we should remove it to: 
1. reduce duplication of information: placeholder + input mask should be good enough.
1. reduce the height of the input takes by default. It might not fit into people's layout.
1. encourage the usage of the helper text for related information regarding the field.

## Actual behavior

<img width="272" alt="Capture d’écran 2020-05-06 à 21 17 55" src="https://user-images.githubusercontent.com/3165635/81218814-1588fe80-8fdf-11ea-9a31-f0db90e58ba6.png">

**Comment by dmtrKovalenko** [comment]
The feature itself is pretty nice and adds really good UX for input. Because when you type 01 you don’t have any information about the format you need to type. 

And in order to achieve it we need to make another prop ‘enableAutoHelperText’ for example. Because this prop is auto generated on the library side. 

It is much easier to override and disable this prop than include it. And it provides much better UX

**Comment by oliviertassinari** [comment]
@dmtrKovalenko Ok thanks for sharing the incentive. What do you think of we wait for the adoption of the next version to see more feedback?

**Comment by dmtrKovalenko** [comment]
Agree, let's wait for the reactions. 

One more thing I want to mention. That for most of the cases (when `label` is used) there is no duplication. Here is the most used UX: 

![Screen Recording 2020-05-06 at 22 44 17](https://user-images.githubusercontent.com/16926049/81221315-5555e300-8feb-11ea-92ba-77159601844e.gif)

And it is possible to extend or override helperText like
```jsx
renderInput={props => 
  <TextField {...props} helperText={`date format: ${props.helperText}`} /> 
} 
```

**Comment by dmtrKovalenko** [comment]
Yes, let’s wait for reactions. 
One more thing I

**Comment by jvhellemondt** [comment]
I do agree with:

> The feature itself is pretty nice and adds really good UX for input. Because when you type 01 you don’t have any information about the format you need to type.
> ...
> Conclusion: It is much easier to override and disable this prop than include it. And it provides much better UX

It is an opinion though.

Anyway, in my use case, I did "helperText={null}". Not because I don't like the helperText, it doesn't fit in my layout (it should be in one line, and the line get's like +12px on height with the helperText).

**Comment by oliviertassinari** [comment]
> It is much easier to override and disable this prop than include it.

@dmtrKovalenko What's harder between?

```jsx
renderInput={props => 
  <TextField {...props} helperText={null} /> 
} 
```

and 

```jsx
renderInput={props => 
  <TextField {...props} helperText={props.placeholder} /> 
} 
```

**Comment by jvhellemondt** [comment]
@oliviertassinari 
I support keeping the helpertext (having a value by default) in: it's better to notice it being there and disable it - than forgetting about it and possibly have a worse UI/UX. The question is, how opinionated do you want to have it.

**Comment by BjoernRave** [comment]
+1 for at least having an option to disable the helpertext completely

**Comment by taschetto** [comment]
I'm experimenting with 4.0.0-alpha.7 and it's very good. I understand the purpose on showing the format by default, but my users are all on the same locale and we have a pretty strong date format default. So the helperText with the format is undesirable. +1 for at least having an option to disable.

**Comment by dmtrKovalenko** [comment]
There is an option. Pass `helperText={null}`

**Comment by oliviertassinari** [comment]
Closing as it doesn't seem to get traction.

**Comment by ifndefdeadmau5** [comment]
`helperText={null}` seems very awkward to me, where you can always pass it intentionally to the `<TextField />` when it's needed
I personally think it's unnecessary default behavior

**Comment by oliviertassinari** [comment]
Agree, looking at the date pickers in the wild such helper text seems to be the exception, I think that it will be better to remove it.

**Comment by dmtrKovalenko** [comment]
Hardly disagree, what the problem to keep this feature?  I am not seeing any. 
Trying to be the same as other datepickers – will kill the project. We must be the best.

**Comment by oliviertassinari** [comment]
> Trying to be the same as other datepickers

To be more precise, I was making a reference to the usage of the date picker in developer's applications & website. I think that it's the dimension we should optimize againt: what are our users' UI/UX designers asking for the most often.

**Comment by dmtrKovalenko** [comment]
What if they all are not right? This is the case as for me. Because helper text is the best way to contain information that helps input the right date.

Personally I have a lot of cases when I am starting typing date and after typing the first part realize that I don't understand right format. So I need to delete text and check the placeholder again. Or endup with `31.12.2019` when it was required to type `12.31.2019`.

**Comment by oliviertassinari** [comment]
I think that it's a tradeoff between display density, and ease of filling the field. One could argue that a placeholder, a pop-up, and using the locale format is more than enough. Let's wait a couple more months before leaning on a specific direction. I think that we can keep paying attention to the date pickers present in the wild :).

**Comment by oliviertassinari** [comment]
Let's aim for high density, which will better answer our market.

**Comment by iamhimadri** [comment]
![Screenshot 2020-08-28 at 5 52 19 PM](https://user-images.githubusercontent.com/23113952/91560297-40fd7380-e957-11ea-8dee-bb897cc3927d.png)

+1 helperText should be optional. There should be a way to hide it. It's looking odd in my use-case.

**Comment by dmtrKovalenko** [comment]
@oliviertassinari I tent to agree that it should not be included by default if so many users cannot find a way to hide it. 

My original thought that it will be much clear how to disable it.
